experimental/dds/tree2/src/test/feature-libraries/editable-tree/editableTree.editing.spec.ts

```diff
@@ -662,9 +664,80 @@ describe("editable-tree: editing", () => {
 				field.setContent(undefined);
 			});
 
-			// delete method
+			// remove method
 			assert.throws(() => {
-				field.delete();
+				field.remove();
+			});
+		});
+	});
+
+	describe("treeStatus", () => {
+		describe("EditableTree", () => {
+			it("root node returns TreeStatus.InDocument", () => {
+				const view = createSharedTreeView().schematize({
+					schema: getTestSchema(FieldKinds.sequence),
+					allowedSchemaModifications: AllowedUpdateType.None,
+					initialTree: { foo: [], foo2: [] },
+				});
+				const rootNode = view.root;
+				assert(isEditableTree(rootNode));
+				const rootNodeStatus = rootNode[treeStatus]();
+
+				assert.equal(rootNodeStatus, TreeStatus.InDocument);
+			});
+
+			it("removed node returns TreeStatus.Deleted", () => {
+				const view = createSharedTreeView().schematize({
+					schema: getTestSchema(FieldKinds.sequence),
+					allowedSchemaModifications: AllowedUpdateType.None,
+					initialTree: { foo: ["foo", "bar"], foo2: [] },
+				});
+				const root = view.root;
+				assert(isEditableTree(root));
+				const field = root[localFieldKey];
+				assert(isEditableField(field));
+
+				const node = field.getNode(0);
+				const statusBeforeRemove = node[treeStatus]();
+				assert.equal(statusBeforeRemove, TreeStatus.InDocument);
+
+				field.remove();
+				const statusAfterRemove = node[treeStatus]();
+				assert.equal(statusAfterRemove, TreeStatus.Deleted);
+			});
+		});
+
+		describe("EditableField", () => {
+			it("root field returns TreeStatus.InDocument", () => {
+				const view = createSharedTreeView().schematize({
+					schema: getTestSchema(FieldKinds.sequence),
+					allowedSchemaModifications: AllowedUpdateType.None,
+					initialTree: { foo: [], foo2: [] },
+				});
+				const rootField = view.context.root;
+				const rootNodeStatus = rootField.treeStatus();
+
+				assert.equal(rootNodeStatus, TreeStatus.InDocument);
+			});
+
+			it("removed field returns TreeStatus.Deleted", () => {
+				const view = createSharedTreeView().schematize({
+					schema: getTestSchema(FieldKinds.sequence),
+					allowedSchemaModifications: AllowedUpdateType.None,
+					initialTree: { foo: ["foo"], foo2: [] },
+				});
+				const root = view.root;
+				assert(isEditableTree(root));
+				const field = root[localFieldKey];
+				assert(isEditableField(field));
+				const statusBeforeRemove = field.treeStatus();
+				assert.equal(statusBeforeRemove, TreeStatus.InDocument);
+
+				const rootField = view.context.root;
+				rootField.remove();
+
+				const statusAfterRemove = field.treeStatus();
+				assert.equal(statusAfterRemove, TreeStatus.Deleted);
 			});
```

Comment:
It seems that when I try to call the `treeStatus()` method on a removed field, it asserts error `0x3c4 /* It is invalid to access an EditableTree node which no longer exists */`. Is there a good way to get around this?

Reply:
Look at the call stack and see why its erroring. Its possible you will need to avoid accessing something on a deleated tree thats invalid and/or move where the check for deleted trees happens.

Reply:
Changed the logic order in `get` of `FieldProxyTarget`